### ISSUE TITLE
Make `updateOverlappingItemSkillNames` use gausie's hot new data-of-loathing API, and only trigger when we're building for a release

### ISSUE THREAD

**Issue by spaghetti-squash** [loathers/libram#573]
Right now, `updateOverlappingItemSkillNames` hits kolmafia's github directly, and seems to fire pretty often. Sorry!

The former is dangerous because if kolmafia changes the way it encodes data, we can get build issues--this happened a little while ago. data-of-loathing is more deliberately safeguarded against that. We could just apply those safeguards here, but why would we do that when we could instead rely on this beautiful tool.

The latter isn't dangerous, but it is wasteful.

**Comment by Rinn** [comment]
what if we just forced combat macros to always use ids? we wouldn't need that at all if so.

**Comment by horrible-little-slime** [comment]
> what if we just forced combat macros to always use ids? we wouldn't need that at all if so.

I don't think this is an awful idea, but we in general try to use strings over numbers for casual legibility
